src/features/contracts/contracts-directory.ts

```diff
@@ -177,13 +178,34 @@ export const getContracts = async (type: string): Promise<Contract[]> => {
 		return [];
 	}
 
-	const contractFiles = await glob(
+	let contractFiles = await glob(
 		`${CONTRACTS_BASE_DIR}/**/contracts/${type}/**/*.json`,
 	);
 	if (!contractFiles.length) {
 		return [];
 	}
 
+	// If there are explicit includes, then everything else is excluded so we need to
+	// filter the contractFiles list to include only contracts that are in the INCLUDED_DEVICETYPE_SLUGS map
+	if (INCLUDED_DEVICETYPE_SLUGS.size > 0) {
```

Comment:
If we want to be 100% accurate (even though atm there is no clash, so there isn't any issue), I think this should be:
```suggestion
	if (type === 'hw.device-type' && INCLUDED_DEVICETYPE_SLUGS.size > 0) {
```

Reply:
for my use, I was looking to filter `arch.sw`, `hw.device-manufacturer`, `hw.device-family`, and `hw.device-type` contracts to the absolute minimum set my deployment needed, which is `aarch64;iot-gate-imx8;generic-aarch64;amd64;generic-amd64`. Limiting this to just `hw.device-type` contracts would pull in 4 `arch.sw` contracts I don't need (armv7hf, i386-nlp, i386, and rpi), 2 `hw.device-manufacturer` contracts I don't need (aaeon and asus), and 2 `hw.device-family` contracts I don't need (tinkerboard and upboard).

Reply:
I had originally called the config variable `INCLUDED_CONTRACT_AND_DEVICETYPE_SLUGS`, but it felt way too long - but I can see how leaving it as just `DEVICETYPE_SLUGS` makes the filtering of `arch.sw`, `hw.device-manufacturer`, and `hw.device-family` contract filtering awkward. I tried to make up for it with a good description of behavior in config.ts.

It seemed unlikely that there would be clashes between the various contract folders, and device-type-lists.ts filtering needed just the slug without the contract-type prefix, so I went with bare slugs instead of needing both with something like: `arch.sw/amd64;hw.device-type/generic-arm64;generic-amd64`

Reply:
@thgreasi ?

Reply:
Hi @shaunco and I'm sorry for my late response here.
Just renaming the env var to `INCLUDED_CONTRACT_SLUGS` to make it clearer what it refers to should be enough as you mention. eg:
```
INCLUDED_CONTRACT_SLUGS="amd64;armv7hf;generic-amd64;fincm3"
```
I do realize that having clashes sounds like a hard thing to occur, but to be honest I prefer your seconf suggestion about defining the fully qualified slugs of all the contracts that a user cares to be included, which doesn't leave any room for ambiguities. eg:
```
INCLUDED_CONTRACTS="arch.sw/amd64;arch.sw/armv7hf;hw.device-type/generic-arm64;hw.device-type/fincm3"
```

Reply:
Or maybe `CONTRACTS_ALLOWLIST`, which also gets alphabetically sorted next to all other contract related env vars?

Reply:
Thanks @thgreasi !

I'll go with `CONTRACT_ALLOWLIST`.

For the fully qualified slugs, would you prefer `device-type-lists.ts` to simply strip the `hw.device-type` off of `hw.device-type/generic-amd64` to get the `generic-amd64` slug that it needs, or would you prefer that the env variable must have BOTH `hw.device-type/generic-amd64` AND `generic-amd64` (that is: `CONTRACT_ALLOWLIST="hw.device-type/generic-amd64;generic-amd64"`)?

Reply:
@shaunco In my view it's enough to have just the fully qualified strings and not any device type slugs alone. ie:
```
INCLUDED_CONTRACTS="arch.sw/amd64;hw.device-type/generic-arm64"
```
Then `device-type-lists.ts` would do something along the lines of:
```ts
slugs = slugs.filter((slug) => CONTRACT_ALLOWLIST.has(`hw.device-type/${slug}`));
```
since the string concatenation should be faster than first finding the contract slugs that start with `hw.device-type/`, then slicing that prefix on a second step, and I guess constructing a new Set with all these on a third step to keep the overall iteration of the listed folders an O(N) operation.